lib/active_record_shards/model.rb

```diff
@@ -24,13 +24,22 @@ def is_sharded? # rubocop:disable Style/PredicateName
     end
 
     def on_slave_by_default?
-      if base_class.instance_variable_defined?(:@on_slave_by_default)
-        base_class.instance_variable_get(:@on_slave_by_default)
+      if self == ActiveRecord::Base
+        false
+      else
+        base = base_class
```

Comment:
I’m not sure it’s worth it to set a local variable.

Reply:
idk how expensive base_class is ... I'd wager reusing a local variable is faster then a noop method call ... and `base_class` might do all kinds of checks ... so opting out from that :)

Reply:
Ahh, it’s a method call. I thought it was an attribute. Never mind then :-)